src/main/java/objects/battle/enemy/EnemyFactory.java

```diff
@@ -0,0 +1,142 @@
+package objects.battle.enemy;
+
+import database.managers.AIDataManager;
+import database.managers.EnemyDataManager;
+import database.managers.SkillDataManager;
+import database.objects.AIData;
+import database.objects.EnemyData;
+import database.objects.SkillData;
+import objects.battle.Skill;
+import objects.battle.SkillType;
+import objects.battle.enemy.ai.DefaultAI;
+import objects.battle.enemy.ai.EnemyAI;
+import objects.battle.enemy.ai.SmartAI;
+import objects.battle.enemy.gimmick.*;
+import objects.character.Boss;
+import objects.character.Enemy;
+
+
+import java.util.ArrayList;
+import java.util.Objects;
+
+
+public class EnemyFactory {
+
+    /**
+     * This method returns the enemy instance using the information given by database
+     * @param name of the enemy to create
+     * @return instance of enemy
+     */
+    public static Enemy getEnemy(String name){
+        EnemyDataManager data = new EnemyDataManager();
+        EnemyData enemyData = data.fetchEnemyData(name);
+        ArrayList<Skill> skills = new ArrayList<>();
+        SkillDataManager dataSkill = new SkillDataManager();
+        for (String i:enemyData.skills) {
+            SkillData skillData = dataSkill.fetchSkillData(i);
+            int damage = Integer.parseInt(skillData.damage);
+            String skillType = skillData.type;
+            SkillType type = translateSkill(skillType);
+            int lag = Integer.parseInt(skillData.lag);
+            Skill skill = new Skill(i, damage, lag, type);
+            skills.add(skill);
+        }
+        int reputation = Integer.parseInt(enemyData.reputation);
+        int speed = Integer.parseInt(enemyData.speed);
+
+        String typeStr = enemyData.type;
+        SkillType type = translateSkill(typeStr);
+
+        EnemyInfo enemyInfo = new EnemyInfo(skills, speed, reputation, type);
+
+        EnemyAI AItype;
+        String ai = enemyData.ai;
+        AIDataManager dataAI = new AIDataManager();
+        AIData aiData = dataAI.fetchAIData(ai);
+        if (Objects.equals(ai, "smart")) {
+            AItype = new SmartAI(enemyInfo, Integer.parseInt(aiData.chance));
+        } else{
+            AItype = new DefaultAI(enemyInfo, Integer.parseInt(aiData.chance));
+        }
+        return new Enemy(name, enemyInfo, AItype);
```

Comment:
why use name here? should be based on the display name of the enemy from the database
likely you will be passed the id of the enemy into this method which could be different from the name to be displayed to the user